main/zoe/guide/contracts/autoswap.md

```diff
@@ -2,161 +2,170 @@
+provided than required, we refuse the offer. If more is provided than is required, we
+return the excess.
+
+Before trading can take place, someone must add liquidity using
+`makeAddLiquidityInvitation()`. Separate invitations are available for adding and
+removing liquidity, and for doing swaps. Other API operations support price checks and
+checking the size of the liquidity pool.
+
+The `swap()` operation requires either the input amount or the output amount to be
+specified. `makeSwapInInvitation()` treats the give amount as definitive, while
+`makeSwapOutInvitation()` honors the want amount. With swapIn, a want amount can be
+specified, and if the offer can't be satisfied, the offer will be refunded. Similarly
+with swapOut, the want amount will be satisfied if possible. If more is provided as the
+give amount than necessary, the excess will be refunded. If not enough is provided, the
+offer will be refunded.
+
+The `publicFacet` can make new invitations (`makeSwapInInvitation()`,
+`makeSwapOutInvitation()`, `makeAddLiquidityInvitation()`, and
+`makeRemoveLiquidityInvitation()`), tell how much would be paid for a given input
+(`getInputPrice()`), or how much is required to be deposited in order to get a
+specified amount out (`getOutputPrice()`). In addition, there are requests for the
+Liquidity issuer (`getLiquidityIssuer()`), the current outstanding liquidity
+(`getLiquiditySupply()`), and the current balances in the pool (`getPoolAllocation()`).
 
-An Autoswap is like a swap, except instead of having to find a
-matching offer, an offer is always matched against the existing
-liquidity pool. The Autoswap contract checks whether your offer will
-keep the [constant product
-invariant](https://github.com/runtimeverification/verified-smart-contracts/blob/uniswap/uniswap/x-y-k.pdf)
-before accepting.
-
-An implementation of [UniSwap](https://uniswap.org/).
 
 ## Initialization
 
-Create an instance of the Autoswap code, and receive an invite that
-when used, will let you add liquidity to Autoswap.
+When someone creates an instance of the Autoswap code, they receive only the public
+facet. The creator has no special access or priveleges.
 
 ```js
 const issuerKeywordRecord = harden({
-  TokenA: moolaIssuer,
-  TokenB: simoleanIssuer,
+  Central: moolaIssuer,
+  Secondary: simoleanIssuer,
 });
 
-const addLiquidityInvite = await E(zoe).makeInstance(
-  autoswapInstallationHandle,
+const publicFacet = await E(zoe).startInstance(
+  autoswapInstallation,
   issuerKeywordRecord,
 );
 ```
 
 ## Adding liquidity to the pool
 
-We can contribute to the Autoswap liquidity pool by making an offer by
-using an addLiquidityInvite. For instance, let's say that Alice
-creates a proposal with the associated payments of moola and simoleans
-and escrows them:
+We can contribute to the Autoswap liquidity pool by making an offer using an
+addLiquidityInvitation. For instance, let's say that Alice creates a proposal with the
```

Comment:
`addLiquidityInvitation`

Reply:
It's not intended as the formal term. I split it up more to make it less so.